src/components/AssessmentSetupForm.tsx

```diff
@@ -47,32 +50,55 @@ export const AssessmentSetupForm = () => {
 
     const [companyId, setCompanyId] = React.useState('');
     const [hasCompanyIdError, setHasCompanyIdError] = React.useState(false);
+    const [hasJobRoleError, setHasJobRoleError] = React.useState(false);
     const [assessmentForm, setAssessmentForm] = React.useState({
         ...assessmentFormInitialValues
     });
     const [assessmentFormErrorState, setAssessmentFormErrorState] =
         React.useState({ ...assessmentFormErrorStateInitialValues });
+    const [jobRole, setJobRole] = React.useState('');
 
     const { data: companiesResponse, isLoading: isCompaniesLoading } =
         useGetCompanies();
-    const { companyIdOptions, companyMap } = useCompanyHelper(
-        companiesResponse?.data
+    const { companyIdOptions } = useCompanyHelper(companiesResponse?.data);
+    const {
+        mutate: searchJobRoles,
+        data: jobRolesResponse,
+        isLoading: isJobRolesLoading
+    } = useSearchJobRoles();
+
+    const { jobRoleOptions, jobRoleMap } = useJobRoleHelper(
+        jobRolesResponse?.data
     );
 
     const onCompanyIdChange = ({
         target: { value }
+    }: React.ChangeEvent<HTMLSelectElement>) => {
+        setCompanyId(value);
+        setHasCompanyIdError(!value);
+    };
+
+    const onJobRoleChange = ({
+        target: { value }
     }: React.ChangeEvent<HTMLSelectElement>) => {
         const assessmentSettings =
-            companyMap[value]?.settings.assessmentSettings;
+            jobRoleMap[value]?.settings?.assessmentSettings;
         if (assessmentSettings) {
             const { emails, ...restSettings } = assessmentSettings;
-            setAssessmentForm({ ...restSettings, emails: emails.join(',') });
+            setAssessmentForm({
+                ...restSettings,
```

Comment:
```suggestion
            setAssessmentForm((prev) => {
                ...restSettings,
```

Can we use `prev` to avoid inconsistencies

Reply:
done